src/http-api.ts

```diff
@@ -104,28 +129,83 @@ export class HttpApi {
       }
     });
 
-    this.#api.get('/:did/records/:id', async (req, res) => {
+    this.#api.get('/:did/read/protocols/:protocol/*', async (req, res) => {
+      const query = await RecordsQuery.create({
+        filter: {
+          protocol: req.params.protocol,
+          protocolPath: (req.params[0] || '').replace(leadTailSlashRegex)
+        },
+        pagination: { limit: 1 },
+        dateSort: DateSort.PublishedDescending
+      });
+
+      const { entries, status } = await this.dwn.processMessage(req.params.did, query.message);
+
+      if (status.code === 200) {
+        if (entries[0]) {
+          const record = await RecordsRead.create({
+            filter: { recordId: entries[0].recordId },
+          });
+          const reply = await this.dwn.processMessage(req.params.did, record.toJSON());
+          return readReplyHandler(res, reply);
+        }
+        else {
+          return res.sendStatus(404);
+        }
+      }
+      else if (status.code === 401) {
+        return res.sendStatus(404);
+      }
+      else {
+        return res.sendStatus(status.code);
+      }
+    })
+
+    this.#api.get('/:did/read/protocols/:protocol', async (req, res) => {
```

Comment:
Nit: Write a short comment above each of the endpoints describing it's intended behavior.

At some point in a different effort we could expand on that and find a suitable tool to generate API Endpoint documentation.

Reply:
Alternatively better still (or in addition), just add a `HTTP API` section next to JSON-RPC API section to quickly describe these newly added API (including `query`), dev-rel folks and everyone else will need to know this info anyway.